Document fix and output behaviour for Node.js API usage (#5971)
stylelint/stylelint: docs/user-guide/usage/node-api.md

A string to lint.

> ### `fix`
> 
> Boolean. If `true`, the autofixed code is available as the value of the output property in the returned object.
> 

Reviewer: ```suggestion
```

Let's move this to the [options pages](https://github.com/stylelint/stylelint/blob/main/docs/user-guide/usage/options.md#fix) as it's intentionally deduped.

We can add the line somewhere in the middle, so like:

---

CLI flag: `--fix`

Automatically fix, where possible, problems reported by rules.

For CSS with standard syntax, Stylelint uses [postcss-safe-parser](https://github.com/postcss/postcss-safe-parser) to fix syntax errors.

When using the Node.js API, the autofixed code is available as the value of the `output` property in the returned object.

If a source contains a:

- scoped disable comment, e.g. `/* stylelint-disable indentation */`, any problems reported by the scoped rules will not be automatically fixed anywhere in the source
- unscoped disable comment, i.e. `/* stylelint-disable */`, the entirety of source will not be automatically fixed

This limitation in being tracked in [issue #2643](https://github.com/stylelint/stylelint/issues/2643).

> A string displaying the formatted problems (using the default formatter or whichever you passed).
> 
> If the `fix` option is set to `true`, a string displaying the autofixed code.
> 

Reviewer: ```suggestion
A string that contains either the:

- formatted problems (using the default formatter or whichever you passed)
- or the autofixed code, if the `fix` option is set to `true`

```

Let's make it more explicit that it can contain one of two things.

Note that the customSyntax option also accepts a string. [Refer to the options documentation for details](./options.md#customsyntax).

### Example F

> Using a string and the fix option:

Reviewer: ```suggestion
Using a string and the `fix` option:
```

```js
stylelint
  .lint({
    code: "a { color: pink; }",
    config: { rules: { "hue-degree-notation": "angle" } },
    fix: true
  })
  .then(function () {
    /* .. */
  });
```

> The autofixed code will be available as the value of the output property in the returned object.

Reviewer: ```suggestion
The autofixed code will be available as the value of the `output` property in the returned object.
```
